backend/translation/views.py

```diff
@@ -1,179 +1,212 @@
+        return Response(
+            {"error": "Invalid transcript_id."}, status=status.HTTP_400_BAD_REQUEST
+        )
+
+    # Convert get_latest to boolean
+    get_latest = get_latest == "true"
+
+    # Ensure that required params are present
+    if not (transcript_id and target_lang and translation_type):
+        return Response(
+            {
+                "error": "Missing required query params [transcript_id, target_lang, translation_type]."
+            },
+            status=status.HTTP_400_BAD_REQUEST,
+        )
 
-    @swagger_auto_schema(
-        method="get",
-        manual_parameters=[
-            openapi.Parameter(
-                "transcript_id",
-                openapi.IN_QUERY,
-                description=("A string to pass the transcript uuid"),
+    # Get the translation for the given transcript_id, target_lang and user_id
+    queryset = (
+        Translation.objects.filter(
+            transcript_id=transcript_id,
+            target_lang=target_lang,
+            user=request.user.id,
+            translation_type=translation_type,
+        )
+        .order_by("-updated_at")
+        .first()
+    )
+    # If no translation exists for this user, check if the latest translation can be fetched
+    if queryset is None:
+        if get_latest:
+            queryset = (
+                Translation.objects.filter(
+                    transcript_id=transcript_id, target_lang=target_lang
+                )
+                .order_by("-updated_at")
+                .first()
+            )
+        else:
+            queryset = None
+
+    # If queryset is empty, return appropriate error
+    if not queryset:
+        return Response(
+            {
+                "error": "No translation found for the given transcript_id and target_lang."
+            },
+            status=status.HTTP_404_NOT_FOUND,
+        )
+
+    # Serialize and return the data
+    serializer = TranslationSerializer(queryset)
+    return Response(serializer.data)        
+
+@swagger_auto_schema(
+    method="post",
+    request_body=openapi.Schema(
+        type=openapi.TYPE_OBJECT,
+        required=["target_lang", "captions"],
+        properties={
+            "translation_id": openapi.Schema(
+                type=openapi.TYPE_STRING,
+                description="An integer identifying the translation instance",
+            ),
+            "target_lang": openapi.Schema(
                 type=openapi.TYPE_STRING,
-                required=True,
+                description="A string to pass the target language of the translation",
             ),
-            openapi.Parameter(
-                "target_lang",
-                openapi.IN_QUERY,
-                description=("A string to pass the target language of the translation"),
+            "captions": openapi.Schema(
                 type=openapi.TYPE_STRING,
-                required=True,
+                description="A string to pass the translated captions",
```

Comment:
In the corresponding transcript endpoint, we are calling it `payload`. Shall we follow the same here? Or shall we rename it there?

Reply:
Yeah, we should make it consistent.

Reply:
@pranav2305 the save_translation endpoint now has a request body key `payload` instead of `captions` so that might have to be changed.